### DISCUSSION TITLE
Is there a way to define cells by cell color?

### DISCUSSION THREAD

**Discussion by YeHW** [jsvine/pdfplumber#601] [Q&A] [answered]
I am trying to extract a table like this (black boxes are texts):
![img](https://user-images.githubusercontent.com/43214065/153376896-690cb2ac-b9f2-4a26-9e6d-7caaa59a8a99.png)

I tried a few basic table_settings, but unfortunately, all failed to discern the cells properly.
The default behaivor only recognizes the white horizontal lines while ignoring the vertical separating lines.
I am wondering if there is a way that supports defining cells by their color?

Another caveat is that some of the cells may expand to double of the shown height if the length of text in those cells exceeds the cell width. In other words, the width of each cell is set.

Any ideas that may help resolve the task are welcome.

**Comment by Larbo53** [comment]
Hello,

did you try to indicate the position of the vertical lines as below, then filter by an identical value for each line?
In my case, I get lines from an invoice where for each of them there is a VAT code, for example 'G2'or 'G3'. (see the code below).
Browse the discussions, there may be solutions.

table=doc.pages[l].extract_table(
table_settings={
"vertical_strategy": "explicit",
"explicit_vertical_lines": vertical_lines,
"horizontal_strategy": "text",
"intersection_x_tolerance": 5}

where
vertical_lines
['18.3', '71.3', '171.3', '329.3', '363.4', '446.3', '500', '553.4', '580']

**Comment by jsvine** [comment, marked as answer]
Hi @YeHW, and thanks for your interest in this library. Unfortunately, it is very difficult to help with a PDF based only on a screenshot, since it describes very little of the actual underlying structure of the PDF. Are you able to share that, or a version redacted via https://github.com/JoshData/pdf-redactor?

>  I am wondering if there is a way that supports defining cells by their color?

There is not currently a way to incorporate rectangle color information in `.extract_text()`, but you could write custom code to examine the `non_stroking_color` attributes of all `page.rects` objects, and then use that information to inform your table-extraction strategy.
